packages/visual-editor/src/components/Dropzone/Dropzone.tsx

```diff
@@ -125,6 +128,22 @@ export function Dropzone({
     node?.data.blockId
   );
 
+  // To avoid a circular dependency, we create the recursive rendering function here and trickle it down
+  const renderDropzone = useCallback(
```

Comment:
could this be moved to its own hook to keep DropZone more cohesive?

Reply:
If we have a separate file the renders the `Dropzone` we get again a circular dependency. I wasn't able to get rid of this warning than doing it in the same file 🤷‍♂️

Reply:
Ahh, I see. The `renderDropZone` function uses a `DropZone` itself, causing a circular dep between the two. Oh well, it can't all be perfect 😹 .